src/cdk/v2/destinations/reddit/procWorkflow.yaml

```diff
@@ -0,0 +1,121 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+    exportAll: true
+  - path: ./config
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+  - name: isAppleFamily
+    path: ../../../../v0/util/index
+
+steps:
+  - name: validateInput
+    template: |
+      let messageType = .message.type;
+      $.assert(.message.type, "message Type is not present. Aborting message.");
+      $.assert(.message.type.toLowerCase() ==='track', "Event type " + .message.type.toLowerCase() + " is not supported. Aborting message.");
+      $.assert(.message.event, "Event is not present. Aborting message.");
+
+  - name: messageType
+    template: |
+      .message.type.toLowerCase()
+
+  - name: prepareTrackPayload
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    steps:
+      - name: prepareUserObject
+        template: |
+          const os = .message.context.os.name;
+          let user = .message.().({
+            "email": $.SHA256({{{{$.getGenericPaths("email")}}}}),  
+            "external_id": $.SHA256({{{{$.getGenericPaths("userId")}}}}),
+            "ip_address": $.SHA256(.context.ip || .request_ip),
+            "uuid": .properties.uuid,
+            "user_agent": .context.userAgent,
+            "idfa": $.isAppleFamily(os)? $.SHA256(.context.device.advertisingId): null,
+            "aaid": os?.toLowerCase() ==="android"? $.SHA256(.context.device.advertisingId): null,
+            "opt_out": .properties.optOut,
+            "screen_dimensions": {"width": .context.screen.width, "height":  .context.screen.height},
+          });
+          $.removeUndefinedAndNullValues(user)
+
+      - name: customFields
+        template: |
+          const customFields = .message.().({
+            "currency": .properties.currency,
+              "value": .properties.revenue !== undefined ? Number(.properties.revenue) : undefined,
+            "item_count": .properties.itemCount && Number(.properties.numOfItems),
+            "value_decimal": .properties.revenue !== undefined ? Number(.properties.revenue)/100 : undefined,
+            "conversion_id": .properties.conversionId || .messageId,
+          });
+          $.removeUndefinedAndNullValues(customFields)
+
+      - name: eventType
+        template: |
+          let event = .message.event;
+          let eventInLowerCase = event.toLowerCase();
+          let eventNames = .destination.Config.eventsMapping.(){.from === event}.to[] ?? [];
+          eventNames.length === 0 ? eventNames = $.ecomEventMaps.(){eventInLowerCase in .src}.dest[] ?? [];
```

Comment:
eventNames.length === 0 ? eventNames = $.ecomEventMaps.(){eventInLowerCase in .src}.dest[] : [];
?? -> :